docs/DeveloperGuide.md

```diff
@@ -1,29 +1,739 @@
+This `Command` object will be executed in `BudgetBuddy`. The `Command` object 
+utilizes methods and the classes present in `Commons`, which will be explained in more 
+detail in the following sections.
+
+#### 3.2 Parser Class
+The main functionality of the Parser Class is to determine the type of `CommandCreator` object to initialize. Using
+Boolean Functions, the Parser Class determines this by what the user input starts with. 
+After determining the type of `CommandCreator` object, the Parser initializes the `CommandCreator` object
+with all its required parameters. 
+
+Here are some examples :
+
+| Boolean Method        | Checks if input starts with | Feature Requires   | Creates                                    |
+|-----------------------|-----------------------------|--------------------|--------------------------------------------|
+| isAddExpenseCommand() | add expense                 | input, ExpenseList | AddExpenseCommandCreator(input, expenses)  |
+| isEditSavingCommand() | edit expense                | input, SavingList  | EditSavingsCommandCreator(input, savings)  |
+
+#### 3.3 Ui Class
+The Ui Class is used to print certain elements to the CLI. In particular, it consists of the Welcome Message,
+Goodbye Message, Divider Lines and all the corresponding commands' command format.
+
+#### 3.5 CommandCreator Class
+The CommandCreator class has multiple subclasses, which corresponds to a specific function of the application.
+Within the CommandCreator classes, it handles making sense of the user input, obtaining the relevant parameters, and finally
+creating the `Command` class.
+
+The superclass `CommandCreator` is an abstract class which is never instantiated. Where its createCommand() method is
+overridden by its subclasses.
+
+The association between the `Command` and `CommandCreator` can be seen in their names. E.g. `MenuCommandCreator`, would
+create a `MenuCommand` class when its createCommand() method is called. Similarly, `FindExpensesCommandCreator` would
+create a `FindCommand` class when its createCommand() method is called.
+
+#### 3.4 Command Class
+The Command class, similar to the CommandCreator class, contains multiple subclasses, all corresponding to a specific
+function of the application. Stated in section`3.5 CommandCreator Class`, each subclass of the `Command` Object
+is created by its associated `CommandCreator`. 
+
+The superclass `Command` is an abstract class which is never instantiated. Where its execute() method is overridden
+by its subclasses. What each Command class does when its execute() method is called would be discussed in more detail
+in the Implementation section.
+
+The following UML Sequence Diagram depicts the process of the process through which an input is gone through the application, up till the point
+where the command gets executed :
+![UML Sequence Diagram of Command](diagrams/sequence_diagram_command.jpg)
+
+#### 3.5 Storage Class
```

Comment:
The Storage class can use some further information for developers. Information such as how the save/load function is implemented, how the data is formatted in the output files when writing and how the data is interpreted when reading.